### ISSUE TITLE
Allow records to be used as metadata views

### ISSUE THREAD

**Issue by kzu** [microsoft/vs-mef#343]
Currently, if you try to use a record type as a metadata view, you get the following error:

```
System.NotSupportedException: 'The type [TYPE] is an unsupported type of metadata view.'
```

Given how convenient and concise they are, it would be quite handy if the export provider 
supported record types as metadata views. It would just need to invoke the constructor 
and any init properties as needed.

**Comment by AArnott** [comment]
I can't think of a reason why this wouldn't be technically possible. There may be one, but it's a fine idea to pursue. This doesn't seem to unblock any critical scenario though, so as a low priority item, it won't likely get attention from the VS team. If you're interested in submitting a PR, I'd be willing to review and merge it, if it came with adequate automated tests.

**Comment by AArnott** [comment]
A clarification in light of the metadata-view support added since this issue was filed: non-positional record classes already work when they have a public parameterless constructor and public init properties. They are handled by the existing class metadata-view provider, which constructs the view and assigns those properties. For example, a shape like 

```cs
record HandlerMetadata { public string Name { get; init; } }
```

is already supported. Record structs are not supported by that path.

Positional record classes remain unsupported because their primary constructor must be supplied with metadata values, and that introduces constructor-parameter binding semantics that VS MEF does not currently define.

Are you still interested in this feature? And were you thinking specifically of positional records such as this?

```cs
record HandlerMetadata(string Name, bool SupportsAsync)
```
